GVRf/Framework/framework/src/main/java/org/gearvrf/scene_objects/GVRTextViewSceneObject.java

```diff
@@ -181,6 +205,87 @@ public GVRTextViewSceneObject(GVRContext gvrContext) {
         this(gvrContext, DEFAULT_TEXT);
     }
 
+
+    /**
+     * Special constructor called by X3D parser
+     * setTextPlaneSize() called once parameters such as font and size are set.
+     *
+     * @param gvrContext
+     * @param strings
+     */
+    public GVRTextViewSceneObject(GVRContext gvrContext, String[] strings) {
+        super(gvrContext);
+
+        final GVRActivity activity = gvrContext.getActivity();
+        mTextView = new TextView(activity);
+        mTextView.setBackgroundColor(Color.TRANSPARENT);
+        mTextView.setTextColor(Color.WHITE);
+        setNumberLines( strings.length );
+        String text = "";
+        for (int i = 0; i < strings.length; i++) {
+            if (i > 0) text += "\n";
+            text += strings[i];
+        }
+        mTextView.setText(text);
+
+        mTextViewContainer = new LinearLayout(activity);
+        mFrameListener = new GVRDrawFrameListenerImpl(this);
+
+
+        GVRTexture texture = new GVRExternalTexture(gvrContext);
+
+        gvrMaterialX3D = new GVRMaterial(gvrContext, GVRShaderType.OES.ID);
+        gvrMaterialX3D.setMainTexture(texture);
+
+        mSurfaceTexture = new SurfaceTexture(texture.getId());
+        mSurface = new Surface(mSurfaceTexture);
+
+        setTextStrings(strings);
+
+        sReferenceCounter++;
+        mCount = sReferenceCounter;
+        mFirstFrame = true;
+    }
+
+    /**
+     * Once the font family, size and style are set, one can build the Quad
+     * based on the width and height of the desired Quad.
+     * @param canvasWidth
+     * @param canvasHeight
+     */
+    public void setTextPlaneSize(float canvasWidth, float canvasHeight) {
+        int canvasWidthInt = (int)Math.ceil(canvasWidth);
+        int canvasHeightInt = (int)Math.ceil(canvasHeight);
+        mTextView.setVisibility(View.VISIBLE);
+        mTextView.setLayoutParams(new LayoutParams( canvasWidthInt,  canvasHeightInt) );
+
+        float width = canvasWidth / FACTOR_IMAGE_SIZE;
+        float height = canvasHeight / FACTOR_IMAGE_SIZE;
+
+        GVRContext gvrContext = getGVRContext();
+        GVRMesh gvrMesh = gvrContext.createQuad(width, height);
+
+        GVRRenderData gvrRenderData = new GVRRenderData(gvrContext);
+        gvrRenderData.setRenderingOrder(GVRRenderData.GVRRenderingOrder.TRANSPARENT);
```

Comment:
If X3D needs the object being transparent assumption, it should just make the call from the parser instead of embedding it in what could be generic otherwise method.